tests/tests/write_texture.rs

```diff
@@ -51,9 +51,16 @@ fn write_texture_subset_2d() {
 
         ctx.queue.submit(None);
 
-        let read_buffer = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+        let read_buffer_0 = ctx.device.create_buffer(&wgpu::BufferDescriptor {
```

Comment:
would imho be nice to parameterize this test to have it  both do a full update and two piecewise.
The second one can then point to this PR, pointing out that it protects against regressing this

Reply:
I can probably trigger the validation error with a simpler test case.
Will try to do that instead as opposed to modifying this one.